test/unit/org/apache/cassandra/index/IndexNameTest.java

```diff
@@ -0,0 +1,119 @@
+
+import org.apache.cassandra.cql3.CQLTester;
+import org.apache.cassandra.exceptions.ConfigurationException;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+@RunWith(Parameterized.class)
+public class IndexNameTest extends CQLTester {
+    @Parameterized.Parameter()
+    public String createIndexQuery;
+
+    @Parameterized.Parameters(name = "{0}")
+    public static List<Object[]> parameters() {
+        return List.of(
+                new Object[]{"CREATE INDEX %s ON %s(%s)"},
+                new Object[]{"CREATE CUSTOM INDEX %s ON %s(%s) USING 'org.apache.cassandra.index.sasi.SASIIndex'"},
+                new Object[]{"CREATE CUSTOM INDEX %s ON %s(%s) USING 'StorageAttachedIndex'"}
+        );
+    }
+
+    private String intoColumnDefs(String[] columnNames) {
+        StringBuilder sb = new StringBuilder();
+        for (String columnName : columnNames) {
+            sb.append(columnName).append(" int,");
+        }
+        return sb.toString();
+    }
+
+    @Test
+    public void testQuotedAndLongColumnNames() {
+        String[] columnNames
+                = new String[]{"\"user name\"",
+                "\"userCountry\"",
+                "\"user-age\"",
+                "\"/user/age\"",
+                "\"userage\"",
+                "\"a very very very very very very very very long field\"",
+                "\"   a_very_very_very_very_very_very_very_very_"
+                        + "very_very_very_very_very_very_very_very_very_"
+                        + "very_very_very_very(very)very_very "
+                        + "_very_very_very_very_very_very_very_"
+                        + "very_very_very_very_very_very_very_very_"
+                        + "very_very_very_very_very_very_very_very_very \"",
+                "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaab"
+        };
+
+        createTable("CREATE TABLE %s (key int,"
+                + intoColumnDefs(columnNames)
+                + "PRIMARY KEY (key))");
+        for (String columnName : columnNames)
+            createIndex(String.format(createIndexQuery, "", "%s", columnName));
+
+        for (int i = 0; i < columnNames.length; i++)
+            execute("INSERT INTO %s (key, " + columnNames[i] + ") VALUES (" + i + ", " + i + ')');
+        for (int i = 0; i < columnNames.length; i++)
+            assertRows(execute("SELECT key, " + columnNames[i] + " FROM %s WHERE " + columnNames[i] + " = " + i), row(i, i));
+
+        flush();
+
+        for (int i = 0; i < columnNames.length; i++)
+            assertRows(execute("SELECT key, " + columnNames[i] + " FROM %s WHERE " + columnNames[i] + " = " + i), row(i, i));
```

Comment:
Nit: can be simplified as:
```suggestion
        beforeAndAfterFlush(() -> {
            for (int i = 0; i < columnNames.length; i++)
                assertRows(execute("SELECT key, " + columnNames[i] + " FROM %s WHERE " + columnNames[i] + " = " + i), row(i, i));
        });
```

Reply:
Thank you!